bftengine/src/bftengine/ClientsManager.cpp

```diff
@@ -97,14 +97,18 @@ void ClientsManager::loadInfoFromReservedPages() {
     ConcordAssert(replyHeader->replyLength + sizeof(ClientReplyMsgHeader) <= maxReplySize_);
 
     // YS TBD: Multiple replies for client batching should be sorted by incoming time
-    auto& repliesInfo = indexToClientInfo_.at(clientIdx).repliesInfo;
-    if (repliesInfo.size() >= maxNumOfReqsPerClient_) deleteOldestReply(clientId);
-    const auto& res = repliesInfo.insert_or_assign(replyHeader->reqSeqNum, MinTime);
-    const bool added = res.second;
-    LOG_INFO(CL_MNGR, "Added/updated reply message" << KVLOG(clientId, replyHeader->reqSeqNum, added));
-
+    auto& clientInfo = indexToClientInfo_.at(clientIdx);
+    {
+      std::unique_lock<std::mutex> lock(clientInfo.repliesInfoGuard);
+      auto& repliesInfo = clientInfo.repliesInfo;
+      if (repliesInfo.size() >= maxNumOfReqsPerClient_) deleteOldestReply(clientId);
+      const auto& res = repliesInfo.insert_or_assign(replyHeader->reqSeqNum, MinTime);
+      const bool added = res.second;
+      LOG_INFO(CL_MNGR, "Added/updated reply message" << KVLOG(clientId, replyHeader->reqSeqNum, added));
```

Comment:
Nit: the log could be put out of the lock. Actually, "const bool added = res.second;", too.